### PR TITLE
pmw3360 sensor documentation

### PR DESCRIPTION
**Pull request by zycz**
Change adds documentation related to pmw3360 sensor.

Jira: NCSDK-11558

### CODE REVIEW

File: doc/nrf/drivers/pmw3360.rst [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+The sensor supports a custom set of attributes that are not part of generic sensor attributes defined by Zephyr's :ref:`zephyr:sensor_api` API.
+The attribues are defined as private to the motion sensor in the sensor's header file.
+See API documentation for details.
+
+Usage of burst mode
+===================
+
+Driver uses burst mode for motion read to improve data transfer speed.
+
+
```
Comment (pdunaj) [inline doc/nrf/drivers/pmw3360.rst:129]: excessive empty line

**Review by pdunaj — APPROVED** [review]

**Comment by greg-fer** [comment]
Please also add changelog entry similar to the one from @MarekPieta  for paw3212.

File: doc/nrf/drivers/pmw3360.rst [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+The PMW3360 driver implements a PMW3360 motion sensor driver that is compatible with Zephyr's :ref:`zephyr:sensor_api`.
+
+The sensor driver supports the following features:
+
+* Motion data readouts for X and Y axes
+* :c:enum:`SENSOR_TRIG_DATA_READY` trigger
+* Configuring attributes of the PMW3360 sensor
+
+.. note::
+  The PMW3360 driver is fairly similar to :ref:`paw3212`, but it allows customizing CPI value and downshift time, as detailed in `Kconfig configuration`.
```
Comment (greg-fer) [inline doc/nrf/drivers/pmw3360.rst:19]:
```suggestion
  The PMW3360 driver is similar to :ref:`paw3212`, but it allows customizing CPI value and downshift time, as detailed in `Kconfig configuration`_.
```

File: doc/nrf/drivers/paw3212.rst [hunk 3]
```diff
@@ -15,6 +15,10 @@ The sensor driver supports the following features:
 * :c:enum:`SENSOR_TRIG_DATA_READY` trigger
 * Configuring attributes of the PAW3212 sensor
 
+.. note::
+  The PAW3212 driver is fairly similar to :ref:`pmw3360`, but it allows customizing motion data length instead of CPI and downshift values, as detailed in `Kconfig configuration`.
```
Comment (greg-fer) [inline doc/nrf/drivers/paw3212.rst:19]:
```suggestion
  The PAW3212 driver is similar to :ref:`pmw3360`, but it allows customizing motion data length instead of CPI and downshift values, as detailed in `Kconfig configuration`_.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]
Apologies, spotted two issues in the lines I've added. Could you fix these?

**Review by greg-fer — APPROVED** [review]

File: doc/nrf/drivers/pmw3360.rst [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+   This is done to prevent delaying system start.
+
+   The sensor returns ``-EBUSY`` if the sensor API is used before the asynchronous initialization is completed.
+
+Motion data reading
+===================
+
+Use the following operations to read the motion data:
+
+* Fetch motion data using :c:enum:`SENSOR_CHAN_ALL`.
```
Comment (MarekPieta) [inline doc/nrf/drivers/pmw3360.rst:113]: Actually, we should check the channel in code for consistency with PAW3212 (currently no check for channel value is performed in code).

File: include/sensor/pmw3360.h [hunk 5]
```diff
@@ -6,20 +6,48 @@
...
 extern "C" {
 #endif
 
+/**
+ * @defgroup pmw3360 PMW3360 motion sensor driver
+ * @{
+ * @brief PMW3360 motion sensor driver.
+ */
+
+/** @brief Sensor specific attributes of PAW3212. */
```
Comment (MarekPieta) [inline include/sensor/pmw3360.h:27]: PMW3360 ;)

**Review by MarekPieta — APPROVED** [review]

File: doc/nrf/drivers/pmw3360.rst [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+   This is done to prevent delaying system start.
+
+   The sensor returns ``-EBUSY`` if the sensor API is used before the asynchronous initialization is completed.
+
+Motion data reading
+===================
+
+Use the following operation to read the motion data:
+
+* Read data for motion in the X and Y axes using :c:enum:`SENSOR_CHAN_POS_DX` and :c:enum:`SENSOR_CHAN_POS_DY`, respectively.
```
Comment (MarekPieta) [inline doc/nrf/drivers/pmw3360.rst:113]: You need to fetch the data first. Why did you remove this information ?

**Review by MarekPieta — CHANGES_REQUESTED** [review]

File: doc/nrf/drivers/pmw3360.rst (hunk 6 above)
Comment (zycz) [reply doc/nrf/drivers/pmw3360.rst:113]: Discussed offline.

**Review by MarekPieta — APPROVED** [review]
